index.js

```diff
@@ -220,7 +220,7 @@ class Command extends EventEmitter {
    */
 
   createCommand(name) {
-    return new Command(name);
+    return new this.constructor(name);
```

Comment:
This is not related to the PR. It is an interesting change to simplicity subclassing. I did some quick reading and didn't see drawbacks. Would you like to put that in a separate PR? (Otherwise I'll add it to my list of possible improvements to do myself.)

Reply:
OK, I will move it to separate PR